### PR TITLE
fix: use the test node location when determining snapshot class name

### LINKED ISSUE #195: Shared test mixins that use the snapshot fixture save under wrong name
**Describe the bug**
Shared test mixins that use the snapshot fixture save under the mixin class name rather than the subclass name

**To Reproduce**
```py
class SharedMixin:
    def test_math(self, snapshot):
        assert 1 + 1 == snapshot

class TestMathWorks(SharedMixin):
    pass

class TestMathWorksAgain(SharedMixin):
    pass
```

**Expected behavior**
The snapshot is saved under the subclass names

### PR DESCRIPTION
**Pull request by iamogbz**
## Description

Use the pytest node location to parse the full class name path

## Related Issues

- Closes #195

## Checklist

- [x] This PR has sufficient test coverage.
- [x] I will merge this pull request with a semantic title.

## Additional Comments

No additional comments.

### CODE REVIEW

File: src/syrupy/location.py [hunk 1]
```diff
@@ -16,8 +17,8 @@ def __init__(self, node: Any):
 
     @property
     def classname(self) -> Optional[str]:
-        classes = self._node.obj.__qualname__.split(".")[:-1]
-        return ".".join(classes) if classes else None
+        _, __, qualname = self._node.location
```
Comment (noahnu) [inline src/syrupy/location.py:20]: Where are you seeing documentation for "location"?

Comment (noahnu) [reply src/syrupy/location.py:20]: ah found it. This is "Item" not "Node": https://docs.pytest.org/en/latest/_modules/_pytest/nodes.html#Item

File: src/syrupy/location.py [hunk 2]
```diff
@@ -16,8 +17,8 @@ def __init__(self, node: Any):
...
+        return ".".join(list(self.__valid_ids(qualname))[:-1]) or None
```
Comment (lime-green) [inline src/syrupy/location.py:21]: the `list()` is probably unnecessary

File: src/syrupy/location.py (hunk 1 above)
Comment (noahnu) [reply src/syrupy/location.py:20]: Traced the source code:

https://github.com/pytest-dev/pytest/blob/7d5f5a878598aaa7f3e100ef6a97bcd13c1b408d/src/_pytest/python.py#L328 -> https://github.com/pytest-dev/pytest/blob/7d5f5a878598aaa7f3e100ef6a97bcd13c1b408d/src/_pytest/python.py#L296

Comment (iamogbz) [reply src/syrupy/location.py:20]: I think it's `Function` which inherits from `Item` and for our use `PyobjMixin`
https://docs.pytest.org/en/latest/_modules/_pytest/python.html#Function

With the `location` property from `Node` using the `reportinfo` method defined in `PyobjMixin`

https://github.com/pytest-dev/pytest/blob/7d5f5a878598aaa7f3e100ef6a97bcd13c1b408d/src/_pytest/python.py#L315-L330

**Pull request merged**